hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -690,20 +693,33 @@ public OutputStream createFile(final Path path,
       perfInfo.registerResult(op.getResult()).registerSuccess(true);
 
       AbfsLease lease = maybeCreateLease(relativePath, tracingContext);
-
+      String eTag = op.getResult().getResponseHeader(HttpHeaderConfigurations.ETAG);
+      checkSmallWriteOptimization();
       return new AbfsOutputStream(
           populateAbfsOutputStreamContext(
               isAppendBlob,
               lease,
-              client,
+              clientHandler,
               statistics,
               relativePath,
               0,
+              eTag,
               contextEncryptionAdapter,
               tracingContext));
     }
   }
 
+  /**
+   * Checks and disables the small write optimization if the client is an instance of {@link AbfsBlobClient}
+   * and the small write optimization is enabled in the configuration.
+   */
+  private void checkSmallWriteOptimization() {
```

Comment:
Why check in store class ? Cant this happen in AbfsOutputStreamContext as a method ?

Reply:
Added check in AbfsOuputStreamContext class